### PR TITLE
fix: cors options pass by promise

### PR DESCRIPTION
**Pull request by climba03003**
Resolve: #129 

Changes:
- fix cors options pass by promise
- update typings
- update readme

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
I didn't understand the issue linked, I'm trusting you:(

**Review by zekth — APPROVED** [review]
lgtm

**Comment by zekth** [comment]
> I didn't understand the issue linked, I'm trusting you:(

callback and promise were not handled the same way. Promise was using reply instead of request. See:
https://github.com/fastify/fastify-cors/blob/fe1b50b556332352197b3cc33f526025d058c72d/index.js#L71

https://github.com/fastify/fastify-cors/blob/fe1b50b556332352197b3cc33f526025d058c72d/index.js#L59

**Comment by climba03003** [comment]
> 
> 
> > I didn't understand the issue linked, I'm trusting you:(
> 
> callback and promise were not handled the same way. Promise was using reply instead of request. See:
> 
> https://github.com/fastify/fastify-cors/blob/fe1b50b556332352197b3cc33f526025d058c72d/index.js#L71
> 
> https://github.com/fastify/fastify-cors/blob/fe1b50b556332352197b3cc33f526025d058c72d/index.js#L59

@mcollina zekth explaination is correct for the issue, should we publish as `major` or `patch`? Although, it is a fix, we are changing the value passing in the function.

File: README.md [hunk 1]
```diff
@@ -77,7 +77,6 @@ fastify.register(require('fastify-cors'), (instance) => { (req, callback) => {
     corsOptions = { origin: true }
   }
   callback(null, corsOptions) // callback expects two parameters: error and options
-  }
```
[the remark this answers is not in the document]
Comment (climba03003) [reply README.md:80]: It is an extra bracket.
I change from `() => { () => {} }` to `() => () => {}`

**Comment by mcollina** [comment]
I'll do a patch - this is a fix.

**Pull request merged**